src/Controllers/CategoryController.php

```diff
@@ -30,10 +31,10 @@ public function showCategory(
 	{
 	    /** @var Request $request */
 	    $request = pluginApp(Request::class);
-		
-	    $category = $this->categoryRepo->findCategoryByUrl($lvl1, $lvl2, $lvl3, $lvl4, $lvl5, $lvl6);
+        $sessionService  = pluginApp(SessionStorageService::class);
+        $category = $this->categoryRepo->findCategoryByUrl($lvl1, $lvl2, $lvl3, $lvl4, $lvl5, $lvl6);
         
-        if($category === null)
+        if($category === null || $category->details->first()->lang != $sessionService->getLang())
```

Comment:
Reicht es hier auf das erste Element von `details` zu prüfen oder muss man hier aus allen Details das in der richtigen Sprache raussuchen?

Reply:
die category wird anhand der generierten url geholt, und dort ist dann nur eine details dran.

Passt also.